troposphere/static/js/components/modals/image_version/ImageVersionEditModal.react.js

```diff
@@ -134,20 +135,27 @@ export default React.createClass({
     // there's a risk of the component being re-rendered by the parent.
     // Should probably verify this behavior, but for now, we play it safe.
     onVersionChange: function (e) {
-      this.setState({versionName: e.target.value});
+      let value = e.target.value;
+      this.setState({versionName: value});
     },
 
     onEndDateChange: function (value) {
       this.setState({versionEndDate: value});
     },
     onUncopyableSelected: function (e) {
-      var uncopyable = (e.target.checked);
+      let uncopyable = (e.target.checked);
       this.setState({versionCanImage: uncopyable});
     },
 
-    onImageSelected: function (image_id) {
-      var image = stores.ImageStore.get(image_id);
-      this.setState({versionImage: image});
+    onImageSelected: function (image) {
+      let versionNameError = "";
+      if(image) {
+        versionNameError = this.validateName(image, this.state.versionName) ? "" : "There already exists a version named '"+this.state.versionName+"' in application '"+image.get('name')+"'.";
+      }
```

Comment:
This usage strengthens my case for a function for the `versionNameError`. See Line 182 below.